bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusProvider.java

```diff
@@ -0,0 +1,478 @@
+                    streamReader.start();
+                    this.streamReader = streamReader;
+                }
+                this.tivoSocket = tivoSocket;
+            } else {
+                logger.debug(" connSocketConnect '{}' - socket creation failed to host '{}', port '{}'",
+                        tivoConfigData.getCfgIdentifier(), tivoConfigData.getHost(), tivoConfigData.getTcpPort());
+                return false;
+            }
+
+            return true;
+
+        } catch (UnknownHostException e) {
+            logger.debug(" TiVo '{}' - while connecting, unexpected host error: '{}'",
+                    tivoConfigData.getCfgIdentifier(), e.getMessage());
+        } catch (IOException e) {
+            if (tivoStatusData.getConnectionStatus() != ConnectionStatus.OFFLINE) {
+                logger.debug(" TiVo '{}' - I/O exception while connecting: '{}'", tivoConfigData.getCfgIdentifier(),
+                        e.getMessage());
+            }
+        }
+        return false;
+    }
+
+    /**
+     * {@link doNappTime} sleeps for the period specified by the getCmdWaitInterval parameter. Primarily used to allow
+     * the TiVo time to process responses after a command is issued.
+     * 
+     * @throws InterruptedException
+     */
+    public void doNappTime() throws InterruptedException {
+        TimeUnit.MILLISECONDS.sleep(tivoConfigData.getCmdWaitInterval());
+    }
+
+    public TivoStatusData getServiceStatus() {
+        return tivoStatusData;
+    }
+
+    public void setServiceStatus(TivoStatusData tivoStatusData) {
+        this.tivoStatusData = tivoStatusData;
+    }
+
+    /**
+     * {@link StreamReader} data stream reader that reads the status data returned from the TiVo.
+     *
+     */
+    public class StreamReader extends Thread {
+        private @Nullable BufferedReader bufferedReader = null;
+
+        /**
+         * {@link StreamReader} construct a data stream reader that reads the status data returned from the TiVo via a
+         * BufferedReader.
+         *
+         * @param inputStream socket input stream.
+         * @throws IOException
+         */
+        public StreamReader(InputStream inputStream) {
+            this.setName("OH-binding-" + thingUid + "-" + tivoConfigData.getHost() + "." + tivoConfigData.getTcpPort());
+            this.bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+            this.setDaemon(true);
+        }
+
+        @Override
+        public void run() {
+            try {
+                logger.debug("streamReader {} is running. ", tivoConfigData.getCfgIdentifier());
+                while (!Thread.currentThread().isInterrupted()) {
+                    String receivedData = null;
+                    try {
+                        BufferedReader reader = bufferedReader;
+                        if (reader != null) {
+                            receivedData = reader.readLine();
+                        }
```

Comment:
You will busy loop if reader is null. Please avoid busy loops at all costs.

Reply:
Changed to throw exception if reader is null.